lib/jekyll/post.rb

```diff
@@ -4,7 +4,7 @@ class Post
     include Convertible
 
     # Valid post name regex.
-    MATCHER = /^(.+\/)*(\d+-\d+-\d+)-(.*)(\.[^.]+)$/
+    MATCHER = /^(.+\/)*(\d+-\d+-\d+(?:.\d+:\d+:\d+)?)-(.*)(\.[^.]+)$/
```

Comment:
I don't think this is a good plan – this should be specified in the `date` front matter attribute. Please revert this and use the calculated date from the post's front matter.